src/Aspire.Dashboard/Components/Dialogs/TextVisualizerDialog.razor

```diff
@@ -0,0 +1,80 @@
+        </FluentLabel>
+
+        <div style="margin-left: auto">
+            <FluentMenuButton id="@_openSelectFormatButtonId" slot="end"
+                              ButtonAppearance="Appearance.Stealth"
+                              IconEnd="@(new Icons.Regular.Size24.SlideTextEdit())"
+                              Text="@Loc[nameof(Dialogs.TextVisualizerSelectFormatType)]"
+                              aria-label="@Loc[nameof(Dialogs.TextVisualizerSelectFormatType)]"
+                              OnMenuChanged="@OnFormatOptionChanged">
+                @foreach (var option in _options)
+                {
+                    <FluentMenuItem Id="@option.Id" Disabled="@(!EnabledOptions.Contains(option.Id))" Value="@option.Id">@option.Name</FluentMenuItem>
+                }
+            </FluentMenuButton>
+        </div>
+    </FluentStack>
+</FluentDialogHeader>
+
+<FluentDialogBody>
+    <div class="log-overflow">
+        <div class="log-container" id="test">
+            <Virtualize Items="@GetLines()" ItemSize="20" OverscanCount="100" TItem="StringLogLine">
+                <div class="log-line-row-container">
+                    <div class="log-line-row">
+                        <span class="log-line-area">
+                            <span class="log-line-number text-visualizer-line-number">@context.LineNumber</span>
+                            @if (context.IsFormatted)
+                            {
+                                <span class="@GetLogContentClass()">
+                                    @context.Content
+                                </span>
+                            }
+                            else
+                            {
+                                <span class="log-content">
+                                    @context.Content
+                                </span>
+                            }
+                        </span>
+                    </div>
+                </div>
+            </Virtualize>
+        </div>
+
+        <FluentButton
+            Id="@_copyButtonId"
+            Class="text-visualizer-copy-button"
+            AdditionalAttributes="@FluentUIExtensions.GetClipboardCopyAdditionalAttributes(Content.Text, ControlsStringsLoc[nameof(ControlsStrings.GridValueCopyToClipboard)], ControlsStringsLoc[nameof(ControlsStrings.GridValueCopied)])">
+        <span slot="start">
+            <FluentIcon Class="copy-icon" Style="display:inline; vertical-align: text-bottom" Icon="Icons.Regular.Size16.Copy" Slot="start" />
+            <FluentIcon Class="checkmark-icon" Style="display:none; vertical-align: text-bottom" Icon="Icons.Regular.Size16.Checkmark" Slot="start" />
+        </span>
+
+            @ControlsStringsLoc[nameof(ControlsStrings.GridValueCopyToClipboard)]
+        </FluentButton>
+    </div>
+</FluentDialogBody>
+
+<FluentDialogFooter Visible="false" />
+
+<link rel="stylesheet" href="css/highlight-11.10.0-default.min.css" />
```

Comment:
Linking to a stylesheet inside a component is odd.

I think the content should either be in the TextVisualizerDialog.razor.css file (with a comment to say where it is from) or just add this to the main layout. Could improve in the future.

Reply:
Why can’t we only link inside the component? That way, the stylesheet is conditionally imported

Reply:
CSS link in the page doesn't stop content from being displayed unstyled. There is the danger of a flash of unstyled content when doing this.

It's best practice to place CSS files in the page head so they are loaded before content is displayed.

Reply:
https://stackoverflow.com/questions/1642212/whats-the-difference-if-i-put-css-file-inside-head-or-body

Reply:
I've moved the CSS imports into `MainLayout.razor` because we have to conditionally import a dark or light theme in highlight.js depending on the effective theme. I also extended `ThemeManager` to obtain this effective theme on first render and subsequent theme changes.

Let me know if you can think of a better way